### PR TITLE
Bugfix, improvements to DANDI upload functionality

### PR DESCRIPTION
**Pull request by ttngu207**

### CODE REVIEW

File: element_interface/dandi.py [hunk 1]
```diff
@@ -36,13 +38,17 @@ def upload_to_dandi(
         working_directory, str(dandiset_id)
     )  # enforce str
 
-    download(
-        f"https://gui-staging.dandiarchive.org/#/dandiset/{dandiset_id}"
-        if staging
-        else f"https://dandiarchive.org/dandiset/{dandiset_id}",
-        output_dir=working_directory,
+    dandiset_url = f"https://gui-staging.dandiarchive.org/#/dandiset/{dandiset_id}" if staging else f"https://dandiarchive.org/dandiset/{dandiset_id}/draft"
```
Comment (kabilar) [inline element_interface/dandi.py:41]: Just to confirm, is the `draft` string appended to the URL required when not using the staging server?

File: element_interface/dandi.py [hunk 2]
```diff
@@ -36,13 +38,17 @@ def upload_to_dandi(
...
+
+    subprocess.run(
+        ["dandi", "download", "--download", "dandiset.yaml", "-o", working_directory, dandiset_url],
+        shell=True, 
     )
 
+    # subprocess.run(
+    #     ["nwbinspector", data_directory, "--config", "dandi", "--report-file-path", os.path.join(data_directory, 'nwbinspector.txt')], shell=True
+    # )
+
```
Comment (kabilar) [inline element_interface/dandi.py]: I would suggest removing commented code.

```suggestion
```

**Review by kabilar — CHANGES_REQUESTED** [review]
Thanks @ttngu207!  A couple minor comments above.  Also, please update the version and changelog so that we can make a new release.

File: element_interface/dandi.py (hunk 1 above)
Comment (ttngu207) [reply element_interface/dandi.py:41]: `draft` is still needed for production. 
New data upload will always go to the `draft` version on DANDI archive. 
Users need to explicitly "publish" the `draft` (manual clicks on the DANDI website) to make it a "release" version

Comment (kabilar) [reply element_interface/dandi.py:41]: Thanks for clarifying!

**Review by kabilar — APPROVED** [review]

**Comment by kabilar** [comment]
Version 0.6.1 is released.

**Pull request merged**